src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Vlc.DotNet.Core;
+using Vlc.DotNet.Forms;
+
+namespace Samples.WinForms.MultiplePlayers
+{
+    public partial class Form1 : Form
+    {
+        private readonly List<VlcControl> listOfControls = new List<VlcControl>();
+        private const string StreamParams = ":network-caching=2000";
+        private const string MyVlcLibraryPath = @"C:\Program Files (x86)\VideoLAN\VLC";
+
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private void btnAddPlayer_Click(object sender, EventArgs e)
+        {
+            this.AddPlayer();
+        }
+
+        private void btnRemovePlayer_Click(object sender, EventArgs e)
+        {
+            this.RemovePlayer();
+        }
+
+        private void AddPlayer()
+        {
+            lock (listOfControls)
```

Comment:
I'm not a WinForms expert, but do you really need a lock there? I mean, you only call this method in button event handlers. Aren't these event triggered on the UI thread? That would make the lock unneeded.

Same below

Reply:
I am planning to move the code to async/await so the form does not get totally frozen during player creation and deletion.
I have improved it a lot, although I found that player.EndInit is quite slow an must run on the ui thread.
You will see in the next commit

Reply:
Beware, the library is compatible with very old versions of .net (2.0 as far as I remember). Please don't break the compatibility by introducing async/await there

Reply:
the library keeps being compatible. this change only affects the sample. 
Do you want me to revert the sample?

Reply:
Well, you can't do this:

`

            lock (listOfControls)
            {
                await Task.Run(() =>
                {
                    //
                }
            }
`

but you can do this:

`

            await Task.Run(() =>
            {
                    lock(something)
            }
`

Anyway.

I updated the sample with the asyncronous code in the last commit.